src/bson.c

```diff
@@ -35,6 +35,30 @@
 #define PHONGO_IS_CLASS_INSTANTIATABLE(ce) \
 	(!(ce->ce_flags & (ZEND_ACC_INTERFACE|ZEND_ACC_IMPLICIT_ABSTRACT_CLASS|ZEND_ACC_EXPLICIT_ABSTRACT_CLASS)))
 
+#if PHP_VERSION_ID >= 70000
+# define PHONGO_BSON_STATE_ZCHILD(state) (&((php_phongo_bson_state *)(state))->zchild)
+#else
+# define PHONGO_BSON_STATE_ZCHILD(state) (((php_phongo_bson_state *)(state))->zchild)
+#endif
+
+#if PHP_VERSION_ID >= 70000
+# define PHONGO_BSON_TYPE_WRAPPER(wrapper_ce, zchild) \
+	if (wrapper_ce) { \
+		zval zwrapper; \
+		zend_call_method_with_1_params(NULL, wrapper_ce, NULL, "createfrombsontype", &zwrapper, &(zchild)); \
```

Comment:
I did not create a constant like we had for `BSON_UNSERIALIZE_FUNC_NAME`, as I recall some case where `zend_call_method()` could not find the function when using `"createFromBSONType"`. I wasn't able to determine at the time why that wasn't an issue for `BSON_UNSERIALIZE_FUNC_NAME` (defined as `"bsonUnserialize"`), but I'll check again.